calendar.rb

```diff
@@ -0,0 +1,42 @@
+require 'singleton'
+require_relative "event.rb"
+
+class Calendar
+    include Singleton
+    attr_accessor :events
+    
+    def initialize
+        @events=Hash.new{|h,k| h[k] = h.dup.clear}
```

Comment:
what is meant by `h` and `k` in the code block passed to the `new` of `Hash`. variable name should be meaningful.